Simplified/Audiobooks/AudioBookVendors+Extensions.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  AudioBookVendors+Extensions.swift
+//  SimplyE
+//
+//  Created by Vladimir Fedorov on 03.09.2020.
+//  Copyright © 2020 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+import NYPLAudiobookToolkit
+
+extension AudioBookVendors {
+  
+  /// Vendor tag
+  private var tag: String {
+    "\(FeedbookDRMPublicKeyTag)\(self.rawValue)"
+  }
+  
+  /// UserDefaults key to store certificate date
+  private var validThroughDateKey: String {
+    "\(tag)_validThroughDate"
+  }
+  
+  /// Update vendor's DRM key
+  /// - Parameter completion: Completion
+  func updateDrmCertificate(completion: @escaping () -> ()) {
+    switch self {
+    case .cantook: updateCantookDRMCertificate(completion: completion)
+    }
+  }
+  
+  /// Update Cantook DRM public key
+  ///
+  /// If the key is saved and its saved expiration date is later than today, the function doesn't request a new public key.
+  /// - Parameter completion: Completion
+  private func updateCantookDRMCertificate(completion: @escaping () -> ()) {
+    // Check if we have a valid key
+    if let date = UserDefaults.standard.value(forKey: validThroughDateKey) as? Date, Date() < date {
+      // we have a certificate with a valid date
+      completion()
+      return
+    }
+    
+    // Fetch a new drmKey
+    DPLAAudiobooks.drmKey { (data, date, error) in
+      OperationQueue.main.addOperation {
+        if let error = error {
+          if error is DPLAAudiobooks.DPLAError {
+            Log.error(#file, error.localizedDescription)
+          } else {
+            Log.error(#file, "Could not receive DRM public key, URL: \(DPLAAudiobooks.certificateUrl): \(error.localizedDescription)")
+          }
+          completion()
```

Comment:
In case we are unable to receive a new key, the app shows "not supported" error. A good question what user-friendly message we can return, explaining that DRM server is not responding or that key data from the DRM server is empty. I added `error` parameter to callback and the app will show a readable message when the audiobook is not decoded correctly